_episodes/l2_02_basic_unit_testing.md

```diff
@@ -0,0 +1,293 @@
+3. **Large and Active Community**: Google Test has a large and active community of developers. This means that there is ample support available in terms of documentation, tutorials, forums, and online resources. 
+4. **Mature and Stable**: It is a mature and stable framework that has been used extensively in industry projects and open-source software. 
+5. **Wide Platform Support**: Google Test supports multiple platforms, including Windows, Linux, macOS, and various compilers. It is compatible with popular development environments and build systems, making it suitable for a wide range of C++ projects.
+6. **Flexible and Extensible**: Google Test provides flexibility in test organization and customization. It allows you to structure your tests using test cases and test suites. You can also define custom test fixtures and customise test execution and reporting. Additionally, Google Test can be extended with custom assertion macros and utilities to suit your specific testing needs.
+
+
+
+## 3. Writing Unit test using GoogleTest
+
+For this chapter, we will consider the same example `fibonacci.cpp` that we used in the previous chapter. Also, for understanding how to use GoogleTest, we will simply be converting all the tests that we wrote manually in the same file using GoogleTest framework.
+
+For using GoogleTest in your code, you need to follow the following steps in general.
+
+
+
+### 3.1. Adding the required header files
+
+The first step would be to add the required header files in your program. For Googletest, you would need to add the following line in your code.
+
+```cpp
+#include "gtest/gtest.h"
+```
+
+
+
+### 3.2. Create your tests
+
+The next step would be to to define your test. Googletest uses the following convention in naming various tests.
+
+```cpp
+TEST(TestSuiteName, TestName)
+{
+    // Test logic and assertions
+}
+```
+
+The different parts in the above cell have the following meaning
+
+- `TEST`: This is the macro provided by the Google Test framework to define a test case.
+- `TestSuiteName`: It is the name of the test suite or test fixture. It groups related tests together. It should be a valid C++ identifier.
+- `TestName`: It is the name of the individual test case within the test suite. It should also be a valid C++ identifier.
+- `Test logic and assertions`: This is the body of the test case where you write the actual test code, including any necessary assertions to verify the expected behaviour. The section `Test Assertions` describes more about various assertions available in GoogleTest.
+
+
+
+### 3.3. Initialise GoogleTest framework in your main
+
+For this sub-section, we are assuming that you are writing your own main function (GoogleTest also allows you to run your code without your own main function by linking to a main function provided in `lgest_main`. Please see the next section for more details). You would need to include the following lines in your main function as shown below.
```

Comment:
The next section explains how to use `lgest_main`, but there is no explanation of where it is coming from. I think this deserves some explanation - or a link to a explanation somewhere else. 

Also, I would remove the parenthesis as that sentence is important.

Reply:
Addressed in cf42314.